pyuvdata/uvdata/uvdata.py

```diff
@@ -7010,8 +7011,33 @@ def read(
                         background_lsts=background_lsts,
                     )
                     unread = False
-                except KeyError:
-                    warnings.warn("Failed to read {f}".format(f=filename[file_num]))
+                except KeyError as err:
+                    file_warnings = (
+                        file_warnings
+                        + "Failed to read {f} due to KeyError: {e}\n".format(
```

Comment:
Not a big deal, but this would be shorter if you use f-strings.

Reply:
Good suggestion - I did that!